### PR TITLE
[Fix](inverted index) fix inverted index compound reader memory leak

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_compound_reader.cpp [hunk 1]
```diff
@@ -197,8 +197,12 @@ void DorisCompoundReader::copyFile(const char* file, int64_t file_length, uint8_
 }
 
 DorisCompoundReader::~DorisCompoundReader() {
-    if (_own_index_input) {
-        _CLDELETE(entries)
+    if (!_closed) {
+        try{
+            close();
+        } catch (CLuceneError& err) {
+            LOG(ERROR) << "DorisCompoundReader finalize error:" << err.what();
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_compound_reader.cpp:204]: Will it lost data if just log error

**Review by qidaye — DISMISSED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Comment by qidaye** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Pull request merged**